src/ai/backend/test/testcases/session/graph_dependency_retriever.py

```diff
@@ -0,0 +1,36 @@
+from typing import override
+
+from ai.backend.test.contexts.client_session import ClientSessionContext
+from ai.backend.test.contexts.image import ImageContext
+from ai.backend.test.contexts.session import CreatedSessionMetaContext
+from ai.backend.test.contexts.tester import TestSpecMetaContext
+from ai.backend.test.templates.template import TestCode
+
+
+class DependencyGraphRetriever(TestCode):
+    @override
+    async def test(self) -> None:
+        test_spec = TestSpecMetaContext.current()
+        test_id = test_spec.test_id
+        session = ClientSessionContext.current()
+        image = ImageContext.current()
+        batch_session_meta = CreatedSessionMetaContext.current()
+        second_session_name = f"test-{str(test_id)}"
+
+        try:
+            await session.ComputeSession.get_or_create(
+                image=image.name,
+                name=second_session_name,
+                dependencies=[batch_session_meta.name],
+            )
+
+            result = await session.ComputeSession(name=second_session_name).get_dependency_graph()
+            assert result["session_name"] == second_session_name, (
+                "Session name in the graph should match the test session name"
+            )
+            dependencies = result["depends_on"]
+            assert dependencies[0]["session_name"] == batch_session_meta.name
+            assert dependencies[0]["session_id"] == str(batch_session_meta.id)
+
+        finally:
+            await session.ComputeSession(name=second_session_name).destroy()
```

Comment:
Is it necessary to match this in the TestCode when we tried to guarantee this kind of creation and cleanup guarantee with a wrapper template?

Reply:
It appears that the session created directly in the test code is being removed because it is separate from the session created by the SessionTemplateWrapper.

I think Ideally, the logic for creating and destroying sessions should be encapsulated within a TemplateWrapper, using the dependency graph passed in as a context variable.